var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -386,6 +394,17 @@ def cmake_args(self):
             '-DTrilinos_ENABLE_STK=OFF'
         ])
 
+        if '+dtk' in spec:
+            # The Tpetra flag here will overwrite whatever Tpetra flag was set
+            # to previously. Thus, even if you have ~tpetra, it would still
+            # enable it if DTK is enabled.
+            options.extend([
+                '-DTrilinos_EXTRA_REPOSITORIES:STRING=DataTransferKit',
```

Comment:
TriBITs (Trilinos build system) is going to pick it up as long as the resources puts it in `Trilinos/DataTransferKit`, which it does.